probe-rs/src/flash/error.rs

```diff
@@ -0,0 +1,50 @@
+use thiserror::Error;
+
+use crate::config::FlashRegion;
+use crate::error;
+
+#[derive(Error, Debug)]
+pub enum FlashError {
+    #[error("The execution of '{name}' failed with code {errorcode}")]
+    RoutineCallFailed { name: &'static str, errorcode: u32 },
```

Comment:
What does error code mean here? Could we change this to an enum?

Reply:
I have no clue what it means. ARM docs for pack algorithms are the worst. I am also not sure if this is not implementation defined. It's just known that != 0 means bad. So, no, for now I would keep this an actual number.

Reply:
Then I would suggest changing the error to make it more clear what happened, something like  "Flash algorithm function {name} returned an error code."